sklearn/gaussian_process/gpr.py

```diff
@@ -223,11 +220,25 @@ def obj_func(theta, eval_gradient=True):
         else:
             self.log_marginal_likelihood_value_ = \
                 self.log_marginal_likelihood(self.kernel_.theta)
-
+                
+        #Add sample-dependent noise-estimates      
+        if sample_alpha is None:
+            add_alpha=0.0
+        else:
+            if np.iterable(sample_alpha) \
```

Comment:
Why did you move the validation of alpha here? I think the previous place was better.